### PR TITLE
Year in Music fixes and multi year support

### PR DESCRIPTION
**Pull request by amCap1712**
Over the past year, many components of the codebase have been updated. Update the YIM components so that those can work again when run. Also, update the database schema to support storing multiple year's reports.

## TODO
- [ ] Fix playlist import stuff. Now that troi is part of LB, we can invoke playlist generation directly from it instead of importing text files.

### CODE REVIEW

File: frontend/js/src/year-in-music/2021/ComponentToImage.tsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ function ComponentToImage({ data, entityType, user }: ComponentToImageProps) {
         return canvas.toDataURL("image/png", 1.0);
       })
       .then((image) => {
-        saveAs(image, `${user.name}-top-${entityType}s-2021.png`);
+        saveAs(image, `${user.name}-top-${entityType}s-2022.png`);
```
Comment (MonkeyDo) [inline frontend/js/src/year-in-music/2021/ComponentToImage.tsx:61]: This file is all 2021 in this folder (2 more occurrences below in this file)

File: frontend/js/src/year-in-music/2021/YearInMusic.tsx [hunk 2]
```diff
@@ -409,24 +412,43 @@ export default class YearInMusic extends React.Component<
             <p>You will find in this page:</p>
             <ul>
               <li>
-                {yourOrUsersName} top <a href="#top-releases">albums</a>,{" "}
-                <a href="#top-recordings">songs</a> and{" "}
-                <a href="#top-artists">artists</a> of the year
+                {yourOrUsersName} top{" "}
+                <a href="listenbrainz/webserver/static/js/src/year-in-music#top-releases">
```
Comment (MonkeyDo) [inline frontend/js/src/year-in-music/2021/YearInMusic.tsx:416]: These don't look right as HTML links

File: frontend/js/src/year-in-music/2022/ComponentToImage.tsx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+              type: "image/png",
+              lastModified: new Date().getTime(),
+            }),
+          ];
+          if (navigator.canShare({ files })) {
+            navigator
+              .share({
+                files,
+                title: `${user.name}'s Year In Music 2021`,
+                text: `Your top ${entityType}s of 2021`,
```
Comment (MonkeyDo) [inline frontend/js/src/year-in-music/2022/ComponentToImage.tsx:70]:
```suggestion
                title: `${user.name}'s Year In Music 2022`,
                text: `Your top ${entityType}s of 2022`,
```

Comment (MonkeyDo) [reply frontend/js/src/year-in-music/2022/ComponentToImage.tsx:70]: Although I think I might reimplement this component and pass title and text in props, so not critical

File: frontend/js/src/year-in-music/2022/ComponentToImage.tsx [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    const targetId = `savable-${entityType}-component`;
+    const element = document.getElementById(targetId);
+
+    htmlToImage
+      .toPng(element as HTMLElement, {
+        canvasWidth: element!.offsetWidth,
+        canvasHeight: element!.offsetHeight,
+      })
+      .then((image) => {
+        const fileName = `${user.name}-top-${entityType}s-2021.png`;
```
Comment (MonkeyDo) [inline frontend/js/src/year-in-music/2022/ComponentToImage.tsx:57]:
```suggestion
        const fileName = `${user.name}-top-${entityType}s-2022.png`;
```

File: frontend/js/tests/__mocks__/freshReleasesSitewideData.json [hunk 5]
```diff
@@ -1,15 +1,16 @@
 [
     {
-        "artist_credit_name": "をとは",
+        "artist_credit_name": "\u3092\u3068\u306f",
```
Comment (MonkeyDo) [inline frontend/js/tests/__mocks__/freshReleasesSitewideData.json:3]: Are those unicode changes expected?

**Review by MonkeyDo — COMMENTED** [review]
Frontend bits look good apart from the comments below.
Feel free to ignore the feedback on the 2022 frontend files since I'm basically going to redo the page and share component (as a separate PR methinks).

File: frontend/js/src/year-in-music/2021/YearInMusic.tsx (hunk 2 above)
Comment (amCap1712) [reply frontend/js/src/year-in-music/2021/YearInMusic.tsx:416]: Uh, my IDE must done some automatic refactorings.

Comment (MonkeyDo) [reply frontend/js/src/year-in-music/2021/YearInMusic.tsx:416]: I figured that was probably what happened. Same in the 2022 file by the way

**Comment by mayhem** [comment]
This needs rebasing now that #2285 will be merged, yes?

File: frontend/js/tests/__mocks__/freshReleasesSitewideData.json (hunk 5 above)
Comment (amCap1712) [reply frontend/js/tests/__mocks__/freshReleasesSitewideData.json]: Fixed in 2285 and rebased to include fix.

**Comment by amCap1712** [comment]
@mayhem Yes. That is now merged and this PR has been rebased.

**Pull request merged**